Mathlib/Data/Fintype/Basic.lean

```diff
@@ -141,12 +141,71 @@ theorem codisjoint_left : Codisjoint s t ↔ ∀ ⦃a⦄, a ∉ s → a ∈ t :=
 theorem codisjoint_right : Codisjoint s t ↔ ∀ ⦃a⦄, a ∉ t → a ∈ s :=
   Codisjoint_comm.trans codisjoint_left
 
-section BooleanAlgebra
+instance booleanAlgebra [DecidableEq α] : BooleanAlgebra (Finset α) :=
+  GeneralizedBooleanAlgebra.toBooleanAlgebra
 
-variable [DecidableEq α] {a : α}
+end Finset
 
-instance booleanAlgebra : BooleanAlgebra (Finset α) :=
-  GeneralizedBooleanAlgebra.toBooleanAlgebra
+namespace Mathlib.Meta
+open Lean Elab Term Meta Batteries.ExtendedBinder
+
+/-- Elaborate set builder notation for `Finset`.
+
+* `{x | p x}` is elaborated as `Finset.filter (fun x ↦ p x) Finset.univ` if the expected type is
+  `Finset ?α`.
+* `{x : α | p x}` is elaborated as `Finset.filter (fun x : α ↦ p x) Finset.univ` if the expected
+  type is `Finset ?β`.
```

Comment:
```suggestion
* `{x : β | p x}` is elaborated as `Finset.filter (fun x : β ↦ p x) Finset.univ` if the expected
  type is `Finset ?α`.
```

I understand Kim's confusion. Maybe writing it this way around would empathise more that it is deliberate as written?

Or just `α` and `?α` and brush over the fact that they aren't unified yet?

Reply:
I've chosen the brushing 💇